app/code/Magento/Sales/Model/Order.php

```diff
@@ -706,10 +708,11 @@ private function canCreditmemoForZeroTotal($totalRefunded)
         //case when paid amount is refunded and order has creditmemo created
         $creditmemos = ($this->getCreditmemosCollection() === false) ?
              true : (count($this->getCreditmemosCollection()) > 0);
-        $paidAmtIsRefunded = $this->getTotalRefunded() == $totalPaid && $creditmemos;
-        if (($hasDueAmount || $paidAmtIsRefunded) ||
+        $totalRefunded = $this->getTotalRefunded();
+        $paidAmtIsRefunded = $totalRefunded == $totalPaid && $creditmemos && $this->getTotalDue()!=0 && $totalPaid!=0;
```

Comment:
Does it make sense? 
It seams that `getTotalDue` for zero order will be always `0`.

Reply:
I think we should remove the check for $paidAmtIsRefunded because it return **true** so the function **canCreditmemoForZeroTotal** returns false and so we are unable to generate credit memo for second time in the below scenario
1. We order 2 quantities of zero priced product 
2. Create invoice for both the products
3. Generate credit-memo for one product.
Now the Credit memo button will not appear, but it should because we have not refunded all the products.

I think we should remove $paidAmtIsRefunded, **$this->isAllInvoicedRefunded()** will solve the problem to check if all refunded. 

@swnsma  Please let me know.